app/src/org/commcare/suite/model/graph/Graph.java

```diff
@@ -173,7 +174,10 @@ private void evaluateSeries(GraphData graphData, EvaluationContext context) {
                     }
                     String x = s.evaluateX(refContext);
                     String y = s.evaluateY(refContext);
-                    if (x != null && y != null) {
+                    if (x != null && graphData.getType().equals(GraphUtil.TYPE_BAR)) {
```

Comment:
This looks like it'll never be true for xy graphs, and therefore never draw any points on xy graphs. I'd expect it to be `x != null && (y == null || graphData.getType().equals(GraphUtil.TYPE_BAR))`